service/src/main/java/com/epam/esm/services/service/impl/CertificateServiceImpl.java

```diff
@@ -0,0 +1,197 @@
+        }
+    }
+
+    @Override
+    public List<Certificate> findAll() throws ServiceException {
+        try {
+            return dao.findAll();
+        } catch (DAOSQLException e) {
+            throw new ServiceException(e);
+        }
+    }
+
+    @Override
+    public List<Certificate> findBy(EntityFinder<Certificate> entityFinder) throws ServiceException {
+        try {
+            return dao.findBy(entityFinder);
+        } catch (DAOSQLException e) {
+            throw new ServiceException(e);
+        }
+    }
+
+    @Override
+    public List<Certificate> find(Map<String, String> params) throws ServiceException {
+        finder.newFinder();
+        for (String key : params.keySet()) {
+            try {
+                if (key.contains("sort")) {
+                    finder.sortBy(CertificateSortingParameters.getEntryByParameter(key).getValue(),
+                            parseAscDesc(params.get(key)));
+                } else {
+
+                    CertificateSearchParameters parameter = CertificateSearchParameters.getEntryByParameter(key);
+                    switch (parameter) {
+                        case NAME:
+                            finder.findByName(URLDecoder.decode(params.get(key), StandardCharsets.UTF_8));
+                            break;
+                        case DESCRIPTION:
+                            if (StringUtils.isNotEmpty(params.get(key))) {
+                                finder.findByDescription(URLDecoder.decode(params.get(key), StandardCharsets.UTF_8));
+                            }
+                            break;
+                        case TAGNAME:
+                            if (StringUtils.isNotEmpty(params.get(key))) {
+                                finder.findByTagName(URLDecoder.decode(params.get(key), StandardCharsets.UTF_8));
+                            }
+                            break;
+                        case PRICE_LESS:
+                            try {
+                                finder.findByPriceLess(new BigDecimal(params.get(key)));
+                            } catch (NumberFormatException e) {
+                                throw new BadRequestException(e);
+                            }
+                            break;
+                        case PRICE_MORE:
+                            try {
+                                finder.findByPriceMore(new BigDecimal(params.get(key)));
+                            } catch (NumberFormatException e) {
+                                throw new BadRequestException(e);
+                            }
+                            break;
+                    }
+                }
+            } catch (IllegalArgumentException e) {
+                throw new BadRequestException(e);
+            }
+        }
+        return findBy(finder);
+    }
+
+    @Override
+    public List<Certificate> findByTag(int tagId) throws ServiceException {
+        finder.newFinder();
```

Comment:
initialisation.

Reply:
Can offer as it was before:
finder = new Finder();